x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/whitelist/InternalEqlScriptUtils.java

```diff
@@ -33,6 +36,20 @@
     InternalEqlScriptUtils() {
     }
 
+    public static <T> Boolean multiValueDocValues(Map<String, ScriptDocValues<T>> doc, String fieldName, Predicate<T> script) {
```

Comment:
since the script is only a function of a single field at a time, this means that we don't do cartesian product and can't have multiple fields in an expression, right?

I'm not requesting that function, just confirming

Reply:
Not sure what function your refer to. Will try to answer what I _think_ you are asking.
`multiValueDocValues` is a function for a single field, yes, but through the Predicate `script` we can nest multiple `multiValueDocValues` calls one inside another, making it run a `for` loop inside a `for` loop for that many fields are used in the EQL expression. And this is the carthesian product. A good example of this you can find in [this test](https://github.com/elastic/elasticsearch/pull/74567/files#diff-30e5a2f75ed4deef26926af68378a739247e85d7fd92d1e927985338bf2222fbR464-R467).